src/meta/client/MetaClient.cpp

```diff
@@ -39,8 +41,19 @@ MetaClient::~MetaClient() {
 }
 
 
-void MetaClient::init() {
-    loadDataThreadFunc();
+void MetaClient::waitForMetadReady() {
+    while (!ready_) {
+        if (sendHeartBeat_) {
+            auto ret = heartbeat().get();
+            if (!ret.ok()) {
+                LOG(ERROR) << "Heartbeat failed, status:" << ret.status();
+                std::this_thread::sleep_for(std::chrono::milliseconds(100));
+                continue;
+            }
+        }  // end if
+        loadDataThreadFunc();
```

Comment:
Maybe you need to sleep a while here? If it a MetaClient of graph, and fail to load data, it will try to call listSpaces all the time?

Reply:
Now I have added a sleep. Thanks!